test/Elastic.Apm.Tests/LabelTests.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.Report.Serialization;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class LabelTests
+	{
+		private readonly PayloadItemSerializer _payloadItemSerializer;
+
+		public LabelTests() =>
+			_payloadItemSerializer = new PayloadItemSerializer(new MockConfigSnapshot());
+
+		[InlineData("StrValue")]
+		[InlineData(123)]
+		[InlineData(234.0)]
+		[InlineData(346L)]
+		[InlineData(true)]
+		[InlineData(false)]
+		[Theory]
+		public void SingleLabelOnTransactionTests(object labelValue)
+		{
+			var mockPayloadSender = new MockPayloadSender();
+			using var agent = new ApmAgent(new AgentComponents(payloadSender: mockPayloadSender));
+
+			agent.Tracer.CaptureTransaction("test", "test", t =>
+			{
+				var labelName = "myLabel";
+				SetLabel(t, labelValue, labelName);
+
+				t.Context.InternalLabels.Value.InnerDictionary[labelName].Value.Should().Be(labelValue);
+
+				var jsonString = SerializePayloadItem(t);
+				jsonString.Should().Contain(GetAssertString(labelValue, labelName));
+			});
+		}
+
+		[InlineData("StrValue")]
+		[InlineData(123)]
+		[InlineData(234.0)]
+		[InlineData(346L)]
+		[InlineData(true)]
+		[InlineData(false)]
+		[Theory]
+		public void SingleLabelOnSpanTests(object labelValue)
+		{
+			var mockPayloadSender = new MockPayloadSender();
+			using var agent = new ApmAgent(new AgentComponents(payloadSender: mockPayloadSender));
+
+			agent.Tracer.CaptureTransaction("test", "test", t =>
+			{
+				t.CaptureSpan("testSpan", "test", s =>
+				{
+					var labelName = "myLabel";
+					SetLabel(s, labelValue, labelName);
+					var jsonString = new PayloadItemSerializer(new MockConfigSnapshot()).SerializeObject(s);
```

Comment:
```suggestion
					var jsonString = SerializePayloadItem(s);
```

Reply:
Oh, good catch - this is a leftover from our zoom discussion and replace-all of course did not find this.